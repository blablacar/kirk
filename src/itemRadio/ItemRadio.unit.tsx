import React from 'react'

import { fireEvent, render, screen } from '@testing-library/react'

import { ItemRadio, ItemRadioProps, ItemRadioStatus } from './index'

const defaultProps: ItemRadioProps = {
  label: 'Label',
  name: 'name',
  value: 0,
  className: 'custom-class-name',
  labelTitle: 'Label title',
  data: 'Data',
  dataInfo: 'Data info',
  checked: false,
  disabled: false,
  onChange() {},
  onClick() {},
  status: ItemRadioStatus.DEFAULT,
  key: 0,
}

function createProps(props: Partial<ItemRadioProps> = {}): ItemRadioProps {
  return { ...defaultProps, ...props }
}

describe('ItemRadio', () => {
  it('Should have a label around the input', () => {
    const props = createProps({
      label: 'This is the field label',
      labelTitle: '',
      data: '',
      dataInfo: '',
    })
    render(<ItemRadio {...props} />)
    expect(screen.getByRole('radio', { name: 'This is the field label' })).toBeInTheDocument()
  })

  it('Should display a Loader when the component is in loading status', () => {
    const props = createProps({ status: ItemRadioStatus.LOADING })
    render(<ItemRadio {...props} />)
    expect(screen.getByRole('progressbar')).toBeInTheDocument()
  })

  it('Should call onChange when clicking on the radio', () => {
    const props = createProps({ onChange: jest.fn() })
    render(<ItemRadio {...props} />)
    fireEvent.click(screen.getByRole('radio'))
    expect(props.onChange).toHaveBeenCalledWith({ name: 'name', value: 0 })
  })
})
