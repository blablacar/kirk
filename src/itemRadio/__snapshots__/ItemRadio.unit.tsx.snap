// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ItemRadio Should display a CircleIcon with an innerDisc when the input is checked 1`] = `
.c4 {
  fill: #00AFF5;
}

.c4.kirk-icon-wrapper {
  display: inline-block;
  position: relative;
}

.c4.kirk-icon-shadowed {
  border-radius: 50%;
  box-shadow: 0 0 2px rgba(0,0,0,.3);
}

.c4 .kirk-icon-badge {
  position: absolute;
  bottom: 12px;
  left: 12px;
}

.c2 {
  margin: 0;
  font-weight: 400;
}

.c2.kirk-text-button {
  color: #054752;
  font-size: 16px;
  line-height: 20px;
}

.c2.kirk-text-body,
.c2.kirk-text-bodyStrong {
  color: #708C91;
  font-size: 16px;
  line-height: 20px;
}

.c2.kirk-text-caption {
  color: #708C91;
  font-size: 13px;
  line-height: 16px;
}

.c2.kirk-text-title,
.c2.kirk-text-titleStrong {
  color: #054752;
  font-size: 18px;
  line-height: 20px;
}

.c2.kirk-text-display2 {
  color: #054752;
  font-size: 82px;
  line-height: 82px;
  font-weight: 500;
}

.c2.kirk-text-display1 {
  color: #054752;
  font-size: 30px;
  line-height: 1.06;
  font-weight: 500;
}

.c2.kirk-text-subheader,
.c2.kirk-text-subheaderStrong {
  color: #054752;
  font-size: 22px;
  line-height: 24px;
}

.c2.kirk-text-title,
.c2.kirk-text-titleStrong {
  color: #054752;
  font-size: 18px;
  line-height: 20px;
}

.c2.kirk-text-bodyStrong,
.c2.kirk-text-subheaderStrong,
.c2.kirk-text-titleStrong {
  font-weight: 500;
}

.c1 {
  padding-left: 24px !important;
  padding-right: 24px !important;
  margin-left: 0;
  margin-right: 0;
}

.c1 {
  position: relative;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  padding-top: 16px;
  padding-bottom: 16px;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-flex: 1;
  -ms-flex: 1;
  flex: 1;
  border: 0;
  background: none;
}

.c1.kirk-item--clickable:hover {
  cursor: pointer;
  background: #EDEDED;
}

.c1.kirk-item--hideHoverBackground.kirk-item--clickable:hover {
  background: none;
}

.c1:disabled {
  cursor: default;
}

a.c1 {
  background: none;
  -webkit-text-decoration: none;
  text-decoration: none;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  -webkit-tap-highlight-color: rgba(221,221,221,.4);
  -webkit-touch-callout: none;
}

button.c1 {
  cursor: pointer;
  text-align: left;
  width: 100%;
  font-family: inherit;
  background-color: transparent;
  -webkit-tap-highlight-color: rgba(221,221,221,.4);
  -webkit-touch-callout: none;
}

button:focus:not(.focus-visible).c1 {
  outline: none;
}

.c1 .kirk-item-leftText {
  -webkit-flex: 1;
  -ms-flex: 1;
  flex: 1;
}

.c1 .kirk-item-rightText {
  margin-left: 16px;
  text-align: right;
}

.c1 .kirk-text {
  display: block;
}

.c1 .kirk-item-title--withBody {
  margin-bottom: 4px;
}

.c1 .kirk-item--strikethrough {
  -webkit-text-decoration: line-through;
  text-decoration: line-through;
}

.c1 .kirk-item-title--withButtonAddon {
  position: relative;
}

.c1 .kirk-item-leftAddon,
.c1 .kirk-item-rightAddon {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  min-width: 24px;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c1 .kirk-item-leftAddon {
  margin-right: 16px;
}

.c1 .kirk-item-rightAddon {
  margin-left: 16px;
}

.c1.kirk-item--highlighted .kirk-text-title {
  color: #00AFF5;
}

.c1 .kirk-item-title--withButtonAddon button {
  position: absolute;
  width: 24px;
  height: 24px;
  margin-left: 4px;
  bottom: 0;
}

.c0 input {
  position: absolute;
  -webkit-clip: rect(0,0,0,0);
  clip: rect(0,0,0,0);
}

.c0 {
  cursor: pointer;
  border: 3px solid transparent;
  padding: calc(16px - 3px) auto;
}

.c0.focus {
  border: 3px solid #00AFF5;
}

.c3.spinning {
  -webkit-animation: cqjtzU 1.4s linear infinite;
  animation: cqjtzU 1.4s linear infinite;
}

.c3 circle {
  stroke-width: 6;
  stroke-linecap: round;
}

.c3.spinning circle {
  stroke-dasharray: 187;
  stroke-dashoffset: 0;
  -webkit-transform-origin: center;
  -ms-transform-origin: center;
  transform-origin: center;
  -webkit-animation: hqYbNG 1.4s ease-in-out infinite;
  animation: hqYbNG 1.4s ease-in-out infinite;
}

.c3 circle.outer {
  stroke: #00AFF5;
}

.c3 circle.inner {
  stroke-width: 0;
  fill: #00AFF5;
}

.c3.thin circle {
  stroke-width: 3;
}

.c3.absolute {
  position: absolute;
}

<div
  className="c0"
  disabled={false}
>
  <label
    className="kirk-item kirk-item--clickable kirk-item-radio custom-class-name c1"
  >
    <span
      className="kirk-item-leftText"
    >
      <span
        className="kirk-text kirk-text-title c2"
      >
        Label title
      </span>
      <span
        className="kirk-text kirk-text-body kirk-item-body c2"
      >
        Label
      </span>
    </span>
    <span
      className="kirk-item-rightText"
    >
      <div
        className="kirk-text kirk-text-subheaderStrong kirk-item-title c2"
      >
        Data
      </div>
      <div
        className="kirk-text kirk-text-body kirk-item-body c2"
      >
        Data info
      </div>
    </span>
    <span
      className="kirk-item-rightAddon"
    >
      <input
        checked={true}
        disabled={false}
        name="name"
        onBlur={[Function]}
        onChange={[Function]}
        onClick={[Function]}
        onFocus={[Function]}
        type="radio"
        value={0}
      />
      <svg
        aria-hidden={true}
        className="kirk-icon kirk-icon-circle c3 c4"
        height={24}
        viewBox="0 0 66 66"
        width={24}
        xmlns="http://www.w3.org/2000/svg"
      >
        
        <circle
          className="outer"
          cx="33"
          cy="33"
          fill="none"
          r="30"
        />
        <circle
          className="inner"
          cx="33"
          cy="33"
          r="18"
        />
      </svg>
    </span>
  </label>
</div>
`;

exports[`ItemRadio Should display a Loader when the component is in loading status 1`] = `
.c5 {
  fill: #5DD167;
}

.c5.kirk-icon-wrapper {
  display: inline-block;
  position: relative;
}

.c5.kirk-icon-shadowed {
  border-radius: 50%;
  box-shadow: 0 0 2px rgba(0,0,0,.3);
}

.c5 .kirk-icon-badge {
  position: absolute;
  bottom: 12px;
  left: 12px;
}

.c2 {
  margin: 0;
  font-weight: 400;
}

.c2.kirk-text-button {
  color: #054752;
  font-size: 16px;
  line-height: 20px;
}

.c2.kirk-text-body,
.c2.kirk-text-bodyStrong {
  color: #708C91;
  font-size: 16px;
  line-height: 20px;
}

.c2.kirk-text-caption {
  color: #708C91;
  font-size: 13px;
  line-height: 16px;
}

.c2.kirk-text-title,
.c2.kirk-text-titleStrong {
  color: #054752;
  font-size: 18px;
  line-height: 20px;
}

.c2.kirk-text-display2 {
  color: #054752;
  font-size: 82px;
  line-height: 82px;
  font-weight: 500;
}

.c2.kirk-text-display1 {
  color: #054752;
  font-size: 30px;
  line-height: 1.06;
  font-weight: 500;
}

.c2.kirk-text-subheader,
.c2.kirk-text-subheaderStrong {
  color: #054752;
  font-size: 22px;
  line-height: 24px;
}

.c2.kirk-text-title,
.c2.kirk-text-titleStrong {
  color: #054752;
  font-size: 18px;
  line-height: 20px;
}

.c2.kirk-text-bodyStrong,
.c2.kirk-text-subheaderStrong,
.c2.kirk-text-titleStrong {
  font-weight: 500;
}

.c1 {
  padding-left: 24px !important;
  padding-right: 24px !important;
  margin-left: 0;
  margin-right: 0;
}

.c1 {
  position: relative;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  padding-top: 16px;
  padding-bottom: 16px;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-flex: 1;
  -ms-flex: 1;
  flex: 1;
  border: 0;
  background: none;
}

.c1.kirk-item--clickable:hover {
  cursor: pointer;
  background: #EDEDED;
}

.c1.kirk-item--hideHoverBackground.kirk-item--clickable:hover {
  background: none;
}

.c1:disabled {
  cursor: default;
}

a.c1 {
  background: none;
  -webkit-text-decoration: none;
  text-decoration: none;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  -webkit-tap-highlight-color: rgba(221,221,221,.4);
  -webkit-touch-callout: none;
}

button.c1 {
  cursor: pointer;
  text-align: left;
  width: 100%;
  font-family: inherit;
  background-color: transparent;
  -webkit-tap-highlight-color: rgba(221,221,221,.4);
  -webkit-touch-callout: none;
}

button:focus:not(.focus-visible).c1 {
  outline: none;
}

.c1 .kirk-item-leftText {
  -webkit-flex: 1;
  -ms-flex: 1;
  flex: 1;
}

.c1 .kirk-item-rightText {
  margin-left: 16px;
  text-align: right;
}

.c1 .kirk-text {
  display: block;
}

.c1 .kirk-item-title--withBody {
  margin-bottom: 4px;
}

.c1 .kirk-item--strikethrough {
  -webkit-text-decoration: line-through;
  text-decoration: line-through;
}

.c1 .kirk-item-title--withButtonAddon {
  position: relative;
}

.c1 .kirk-item-leftAddon,
.c1 .kirk-item-rightAddon {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  min-width: 24px;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c1 .kirk-item-leftAddon {
  margin-right: 16px;
}

.c1 .kirk-item-rightAddon {
  margin-left: 16px;
}

.c1.kirk-item--highlighted .kirk-text-title {
  color: #00AFF5;
}

.c1 .kirk-item-title--withButtonAddon button {
  position: absolute;
  width: 24px;
  height: 24px;
  margin-left: 4px;
  bottom: 0;
}

.c0 input {
  position: absolute;
  -webkit-clip: rect(0,0,0,0);
  clip: rect(0,0,0,0);
}

.c0 {
  cursor: pointer;
  border: 3px solid transparent;
  padding: calc(16px - 3px) auto;
}

.c0.focus {
  border: 3px solid #00AFF5;
}

.c4.spinning {
  -webkit-animation: cqjtzU 1.4s linear infinite;
  animation: cqjtzU 1.4s linear infinite;
}

.c4 circle {
  stroke-width: 6;
  stroke-linecap: round;
}

.c4.spinning circle {
  stroke-dasharray: 187;
  stroke-dashoffset: 0;
  -webkit-transform-origin: center;
  -ms-transform-origin: center;
  transform-origin: center;
  -webkit-animation: hqYbNG 1.4s ease-in-out infinite;
  animation: hqYbNG 1.4s ease-in-out infinite;
}

.c4 circle.outer {
  stroke: #5DD167;
}

.c4 circle.inner {
  stroke-width: 0;
  fill: #5DD167;
}

.c4.thin circle {
  stroke-width: 3;
}

.c4.absolute {
  position: absolute;
}

.c3 .visually-hidden {
  position: absolute;
  height: 1px;
  width: 1px;
  overflow: hidden;
  -webkit-clip: rect(1px,1px,1px,1px);
  clip: rect(1px,1px,1px,1px);
  white-space: nowrap;
}

.c3.kirk-loader--inline {
  display: inline-block;
}

.c3.kirk-loader--fullScreen {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  position: fixed;
  top: 0;
  left: 0;
  background-color: rgba(255,255,255,0.85);
  height: 100vh;
  width: 100vw;
  z-index: 4;
}

.c3.kirk-loader--block {
  display: block;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  padding: 16px 0;
}

.c3 .kirk-loader--done {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  background-color: #5DD167;
  color: #FFF;
  border-radius: 50%;
}

<div
  className="c0"
  disabled={false}
>
  <label
    className="kirk-item kirk-item--clickable kirk-item-radio custom-class-name c1"
  >
    <span
      className="kirk-item-leftText"
    >
      <span
        className="kirk-text kirk-text-title c2"
      >
        Label title
      </span>
      <span
        className="kirk-text kirk-text-body kirk-item-body c2"
      >
        Label
      </span>
    </span>
    <span
      className="kirk-item-rightText"
    >
      <div
        className="kirk-text kirk-text-subheaderStrong kirk-item-title c2"
      >
        Data
      </div>
      <div
        className="kirk-text kirk-text-body kirk-item-body c2"
      >
        Data info
      </div>
    </span>
    <span
      className="kirk-item-rightAddon"
    >
      <input
        checked={false}
        disabled={true}
        name="name"
        onBlur={[Function]}
        onChange={[Function]}
        onClick={[Function]}
        onFocus={[Function]}
        type="radio"
        value={0}
      />
      <div
        className="kirk-loader--inline c3"
      >
        <div
          aria-valuetext="Loading"
          role="progressbar"
        >
          <div
            className=""
            style={
              Object {
                "height": "24px",
                "width": "24px",
              }
            }
          >
            <svg
              aria-hidden={true}
              className="kirk-icon kirk-icon-circle spinning c4 c5"
              height={24}
              viewBox="0 0 66 66"
              width={24}
              xmlns="http://www.w3.org/2000/svg"
            >
              
              <circle
                className="outer"
                cx="33"
                cy="33"
                fill="none"
                r="30"
              />
            </svg>
          </div>
        </div>
        <div
          className="visually-hidden"
          role="status"
        >
          Loading
        </div>
      </div>
    </span>
  </label>
</div>
`;

exports[`ItemRadio Should forward its props to the Item component 1`] = `
.c4 {
  fill: #00AFF5;
}

.c4.kirk-icon-wrapper {
  display: inline-block;
  position: relative;
}

.c4.kirk-icon-shadowed {
  border-radius: 50%;
  box-shadow: 0 0 2px rgba(0,0,0,.3);
}

.c4 .kirk-icon-badge {
  position: absolute;
  bottom: 12px;
  left: 12px;
}

.c2 {
  margin: 0;
  font-weight: 400;
}

.c2.kirk-text-button {
  color: #054752;
  font-size: 16px;
  line-height: 20px;
}

.c2.kirk-text-body,
.c2.kirk-text-bodyStrong {
  color: #708C91;
  font-size: 16px;
  line-height: 20px;
}

.c2.kirk-text-caption {
  color: #708C91;
  font-size: 13px;
  line-height: 16px;
}

.c2.kirk-text-title,
.c2.kirk-text-titleStrong {
  color: #054752;
  font-size: 18px;
  line-height: 20px;
}

.c2.kirk-text-display2 {
  color: #054752;
  font-size: 82px;
  line-height: 82px;
  font-weight: 500;
}

.c2.kirk-text-display1 {
  color: #054752;
  font-size: 30px;
  line-height: 1.06;
  font-weight: 500;
}

.c2.kirk-text-subheader,
.c2.kirk-text-subheaderStrong {
  color: #054752;
  font-size: 22px;
  line-height: 24px;
}

.c2.kirk-text-title,
.c2.kirk-text-titleStrong {
  color: #054752;
  font-size: 18px;
  line-height: 20px;
}

.c2.kirk-text-bodyStrong,
.c2.kirk-text-subheaderStrong,
.c2.kirk-text-titleStrong {
  font-weight: 500;
}

.c1 {
  padding-left: 24px !important;
  padding-right: 24px !important;
  margin-left: 0;
  margin-right: 0;
}

.c1 {
  position: relative;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  padding-top: 16px;
  padding-bottom: 16px;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-flex: 1;
  -ms-flex: 1;
  flex: 1;
  border: 0;
  background: none;
}

.c1.kirk-item--clickable:hover {
  cursor: pointer;
  background: #EDEDED;
}

.c1.kirk-item--hideHoverBackground.kirk-item--clickable:hover {
  background: none;
}

.c1:disabled {
  cursor: default;
}

a.c1 {
  background: none;
  -webkit-text-decoration: none;
  text-decoration: none;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  -webkit-tap-highlight-color: rgba(221,221,221,.4);
  -webkit-touch-callout: none;
}

button.c1 {
  cursor: pointer;
  text-align: left;
  width: 100%;
  font-family: inherit;
  background-color: transparent;
  -webkit-tap-highlight-color: rgba(221,221,221,.4);
  -webkit-touch-callout: none;
}

button:focus:not(.focus-visible).c1 {
  outline: none;
}

.c1 .kirk-item-leftText {
  -webkit-flex: 1;
  -ms-flex: 1;
  flex: 1;
}

.c1 .kirk-item-rightText {
  margin-left: 16px;
  text-align: right;
}

.c1 .kirk-text {
  display: block;
}

.c1 .kirk-item-title--withBody {
  margin-bottom: 4px;
}

.c1 .kirk-item--strikethrough {
  -webkit-text-decoration: line-through;
  text-decoration: line-through;
}

.c1 .kirk-item-title--withButtonAddon {
  position: relative;
}

.c1 .kirk-item-leftAddon,
.c1 .kirk-item-rightAddon {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  min-width: 24px;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c1 .kirk-item-leftAddon {
  margin-right: 16px;
}

.c1 .kirk-item-rightAddon {
  margin-left: 16px;
}

.c1.kirk-item--highlighted .kirk-text-title {
  color: #00AFF5;
}

.c1 .kirk-item-title--withButtonAddon button {
  position: absolute;
  width: 24px;
  height: 24px;
  margin-left: 4px;
  bottom: 0;
}

.c0 input {
  position: absolute;
  -webkit-clip: rect(0,0,0,0);
  clip: rect(0,0,0,0);
}

.c0 {
  cursor: pointer;
  border: 3px solid transparent;
  padding: calc(16px - 3px) auto;
}

.c0.focus {
  border: 3px solid #00AFF5;
}

.c3.spinning {
  -webkit-animation: cqjtzU 1.4s linear infinite;
  animation: cqjtzU 1.4s linear infinite;
}

.c3 circle {
  stroke-width: 6;
  stroke-linecap: round;
}

.c3.spinning circle {
  stroke-dasharray: 187;
  stroke-dashoffset: 0;
  -webkit-transform-origin: center;
  -ms-transform-origin: center;
  transform-origin: center;
  -webkit-animation: hqYbNG 1.4s ease-in-out infinite;
  animation: hqYbNG 1.4s ease-in-out infinite;
}

.c3 circle.outer {
  stroke: #00AFF5;
}

.c3 circle.inner {
  stroke-width: 0;
  fill: #00AFF5;
}

.c3.thin circle {
  stroke-width: 3;
}

.c3.absolute {
  position: absolute;
}

<div
  className="c0"
  disabled={false}
>
  <label
    className="kirk-item kirk-item--clickable kirk-item-radio custom-class-name c1"
  >
    <span
      className="kirk-item-leftText"
    >
      <span
        className="kirk-text kirk-text-title c2"
      >
        Label title
      </span>
      <span
        className="kirk-text kirk-text-body kirk-item-body c2"
      >
        Label
      </span>
    </span>
    <span
      className="kirk-item-rightText"
    >
      <div
        className="kirk-text kirk-text-subheaderStrong kirk-item-title c2"
      >
        Data
      </div>
      <div
        className="kirk-text kirk-text-body kirk-item-body c2"
      >
        Data info
      </div>
    </span>
    <span
      className="kirk-item-rightAddon"
    >
      <input
        checked={false}
        disabled={false}
        name="name"
        onBlur={[Function]}
        onChange={[Function]}
        onClick={[Function]}
        onFocus={[Function]}
        type="radio"
        value={0}
      />
      <svg
        aria-hidden={true}
        className="kirk-icon kirk-icon-circle c3 c4"
        height={24}
        viewBox="0 0 66 66"
        width={24}
        xmlns="http://www.w3.org/2000/svg"
      >
        
        <circle
          className="outer"
          cx="33"
          cy="33"
          fill="none"
          r="30"
        />
      </svg>
    </span>
  </label>
</div>
`;
